server/routes/message.route.js

```diff
@@ -43,6 +43,12 @@ router.route('/archive/:messageId')
 router.route('/unarchive/:messageId')
     .put(messageCtrl.unarchive);
 
+router.route('/removeReadAt/:messageId')
+    .get(messageCtrl.removeReadAt);
+
+router.route('/setReadAt/:messageId/:newTime')
```

Comment:
I think the new time should be in the request body, rather than a URL param.

Reply:
I'd prefer that the server generate it's own timestamp, instead of having the client send it. It's a bit less flexible, but "Mark as read" is less responsibility for the client than "Set timestamp arbitrarily"

Reply:
Agreed.